### PR TITLE
fi-749 better error messages when search params don't match

### PR DESCRIPTION
**Pull request by czh-orz**
This PR makes the error messages better when parameters don't match. Now it's in the format of:
clinical-status in  AllergyIntolerance/SMART-AllergyIntolerance-28 (["active"]) does not match clinical-status requested (["inactive"])

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-749
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [X] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: generator/uscore/uscore_generator.rb [hunk 1]
```diff
@@ -1122,34 +1122,37 @@ def create_search_validation(sequence)
           case type
           when 'Period', 'date'
             search_validators += %(
-                value_found = resolve_element_from_path(resource, '#{path_parts.join('.')}') { |date| validate_date_search(value, date) }
-                assert value_found.present?, '#{element} on resource does not match #{element} requested'
+                values_found = resolve_path(resource, '#{path_parts.join('.')}')
+                match_found = values_found.any? { |date| validate_date_search(value, date) }
+                assert match_found, "#{element} in #{sequence[:resource]}/\#{resource.id} (\#{values_found}) does not match #{element} requested (\#{value})"
```
Comment (okeefm) [inline generator/uscore/uscore_generator.rb:1127]: Does it make any sense to take some/all of this matching code and move it to a function? Or at least the assertion failure message, since it's the same for pretty much all of the searches listed here.

Comment (arscan) [reply generator/uscore/uscore_generator.rb:1127]: Stylistically I think the generated code is what matters.  Does this comment apply to what is generated, or this code here?  I bring it up because I'd personally like to avoid the situation where this code is clean but then it generates very sparse tests that are just a bunch of function calls that only get called once in a given sequence (making it hard to follow).

Comment (czh-orz) [reply generator/uscore/uscore_generator.rb:1127]: I can separate the assertion failure message or the validation code in the generator do remove some of the duplication. This won't affect the generated code.

Comment (czh-orz) [reply generator/uscore/uscore_generator.rb]: I updated it so that the type specific stuff is in its own function in the generator.
